File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/mapping_rule.py
# Compilable usage examples for mapping rule operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    MappingRuleCreateRequest,
    MappingRuleSearchQueryRequest,
    MappingRuleUpdateRequest,
    Unset,
)


# region CreateMappingRule
def create_mapping_rule_example() -> None:
    client = CamundaClient()

    result = client.create_mapping_rule(
        data=MappingRuleCreateRequest(
            mapping_rule_id="engineering-group-mapping",
            claim_name="groups",
            claim_value="engineering",
            name="Engineering Group Mapping",
        ),
    )

    print(f"Mapping rule: {result.mapping_rule_id}")
# endregion CreateMappingRule


# region GetMappingRule
def get_mapping_rule_example() -> None:
    client = CamundaClient()

    result = client.get_mapping_rule(mapping_rule_id="rule-123")

    print(f"Mapping rule: {result.name}")
# endregion GetMappingRule


# region SearchMappingRule
def search_mapping_rule_example() -> None:
    client = CamundaClient()

    result = client.search_mapping_rule(
        data=MappingRuleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for rule in result.items:
            print(f"Mapping rule: {rule.name}")
# endregion SearchMappingRule


# region UpdateMappingRule
def update_mapping_rule_example() -> None:
    client = CamundaClient()

    client.update_mapping_rule(
        mapping_rule_id="rule-123",
        data=MappingRuleUpdateRequest(
            claim_name="groups",
            claim_value="senior-engineering",
            name="Senior Engineering Mapping",
        ),
    )
# endregion UpdateMappingRule


# region DeleteMappingRule
def delete_mapping_rule_example() -> None:
    client = CamundaClient()

    client.delete_mapping_rule(mapping_rule_id="rule-123")
# endregion DeleteMappingRule

File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/MappingRule.cs
// Compilable usage examples for mapping rule operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class MappingRuleExamples
{
    #region CreateMappingRule
    // <CreateMappingRule>
    public static async Task CreateMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateMappingRuleAsync(new MappingRuleCreateRequest
        {
            ClaimName = "groups",
            ClaimValue = "engineering",
            Name = "Engineering Group Mapping",
        });

        Console.WriteLine($"Mapping rule: {result.MappingRuleId}");
    }
    // </CreateMappingRule>
    #endregion CreateMappingRule

    #region GetMappingRule

    // <GetMappingRule>
    public static async Task GetMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.GetMappingRuleAsync("rule-123");
        Console.WriteLine($"Mapping rule: {result.Name}");
    }
    // </GetMappingRule>
    #endregion GetMappingRule

    #region SearchMappingRule

    // <SearchMappingRule>
    public static async Task SearchMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchMappingRuleAsync(
            new MappingRuleSearchQueryRequest());

        foreach (var rule in result.Items)
        {
            Console.WriteLine($"Mapping rule: {rule.Name}");
        }
    }
    // </SearchMappingRule>
    #endregion SearchMappingRule

    #region UpdateMappingRule

    // <UpdateMappingRule>
    public static async Task UpdateMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        await client.UpdateMappingRuleAsync("rule-123", new MappingRuleUpdateRequest
        {
            ClaimName = "groups",
            ClaimValue = "senior-engineering",
            Name = "Senior Engineering Mapping",
        });
    }
    // </UpdateMappingRule>
    #endregion UpdateMappingRule

    #region DeleteMappingRule

    // <DeleteMappingRule>
    public static async Task DeleteMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        await client.DeleteMappingRuleAsync("rule-123");
    }
    // </DeleteMappingRule>
    #endregion DeleteMappingRule
}
